[dartdoc_test] implement good logging! (#245)
google/dart-neats: dartdoc_test/test/testdata/run analyze command with verbose flag.txt

Extracting code samples ...
Analyzing code samples ...
No issues found!
No issues found!
this error is caused by generated code.
/Users/takumma/work/gsoc/dart-neats/dartdoc_test/.dartdoc_test/example_0.dart(7..20): Unused import: 'dart:convert'.

> this error is caused by generated code.
> /Users/takumma/work/gsoc/dart-neats/dartdoc_test/.dartdoc_test/example_1.dart(7..20): Unused import: 'dart:convert'.

Reviewer: * Making the path relative to the current working directory, that makes it shorter.
 * And writing then thing about it being ignored in a parenthesis at the end of the same line.

# IF `(7..20)` is line number and colum, then consider:

```suggestion
.dartdoc_test/example_1.dart:7:20: Unused import: 'dart:convert'. (ignored because issue occurs in the generated code)
```

This similar to how line numbers and columns are reported in stack traces, example:
```
#0      Object.noSuchMethod (dart:core-patch:1884:25)
#1      Trace.terse.<anonymous closure> (file:///usr/local/google-old/home/goog/dart/dart/pkg/stack_trace/lib/src/trace.dart:47:21)
#2      IterableMixinWorkaround.reduce (dart:collection:29:29)
#3      List.reduce (dart:core-patch:1247:42)
#4      Trace.terse (file:///usr/local/google-old/home/goog/dart/dart/pkg/stack_trace/lib/src/trace.dart:40:35)
#5      format (file:///usr/local/google-old/home/goog/dart/dart/pkg/stack_trace/lib/stack_trace.dart:24:28)
#6      main.<anonymous closure> (file:///usr/local/google-old/home/goog/dart/dart/test.dart:21:29)
#7      _CatchErrorFuture._sendError (dart:async:525:24)
#8      _FutureImpl._setErrorWithoutAsyncTrace (dart:async:393:26)
#9      _FutureImpl._setError (dart:async:378:31)
#10     _ThenFuture._sendValue (dart:async:490:16)
#11     _FutureImpl._handleValue.<anonymous closure> (dart:async:349:28)
#12     Timer.run.<anonymous closure> (dart:async:2402:21)
#13     Timer.Timer.<anonymous closure> (dart:async-patch:15:15)
```
(stole the stack trace from: https://pub.dev/packages/stack_trace)

# IF `(7..20)` is line numbers, then consider:

```suggestion
.dartdoc_test/example_1.dart#L7-20: Unused import: 'dart:convert'. (ignored because issue occurs in the generated code)
```
I'd suggest:
 * Using `.dart#L7-20` instead of `.dart(7..20)` -- if you checkout links on github they typically use `#L7-20` to anchor a particular range in a file. We could also use:
   * `.dart:#7-20`,
   * `.dart:#7`,
   * `.dart:7-20`, or,
   * `.dart:7`.
